client/image_inspect.go

```diff
@@ -5,28 +5,80 @@ import (
 	"context"
 	"encoding/json"
 	"io"
+	"net/url"
 
 	"github.com/docker/docker/api/types/image"
 )
 
-// ImageInspectWithRaw returns the image information and its raw representation.
-func (cli *Client) ImageInspectWithRaw(ctx context.Context, imageID string) (image.InspectResponse, []byte, error) {
+type imageInspectClientOpt struct {
+	raw        *bytes.Buffer
+	apiOptions image.InspectOptions
+}
+
+type ImageInspectOpt func(*imageInspectClientOpt)
+
+// ImageInspectWithRawResponse instructs the client to additionally store the
+// raw inspect response in the provided buffer.
+func ImageInspectWithRawResponse(raw *bytes.Buffer) ImageInspectOpt {
+	return func(opts *imageInspectClientOpt) {
+		opts.raw = raw
+	}
+}
+
+// ImageInspectWithOpts sets the API options for the image inspect operation.
+func ImageInspectWithOpts(opts image.InspectOptions) ImageInspectOpt {
+	return func(clientOpts *imageInspectClientOpt) {
+		clientOpts.apiOptions = opts
+	}
+}
+
+// ImageInspect returns the image information.
+func (cli *Client) ImageInspect(ctx context.Context, imageID string, inspectOpts ...ImageInspectOpt) (image.InspectResponse, error) {
 	if imageID == "" {
-		return image.InspectResponse{}, nil, objectNotFoundError{object: "image", id: imageID}
+		return image.InspectResponse{}, objectNotFoundError{object: "image", id: imageID}
+	}
+
+	var opts imageInspectClientOpt
+	for _, opt := range inspectOpts {
+		opt(&opts)
 	}
-	serverResp, err := cli.get(ctx, "/images/"+imageID+"/json", nil, nil)
+
+	query := url.Values{}
+	if opts.apiOptions.Manifests {
+		if err := cli.NewVersionError(ctx, "1.48", "manifests"); err != nil {
+			return image.InspectResponse{}, err
+		}
+		query.Set("manifests", "1")
+	}
+
+	serverResp, err := cli.get(ctx, "/images/"+imageID+"/json", query, nil)
 	defer ensureReaderClosed(serverResp)
 	if err != nil {
-		return image.InspectResponse{}, nil, err
+		return image.InspectResponse{}, err
 	}
 
-	body, err := io.ReadAll(serverResp.body)
-	if err != nil {
-		return image.InspectResponse{}, nil, err
+	buf := opts.raw
+	if buf == nil {
+		buf = &bytes.Buffer{}
+	}
+
+	if _, err := io.Copy(buf, serverResp.body); err != nil {
+		return image.InspectResponse{}, err
 	}
 
 	var response image.InspectResponse
-	rdr := bytes.NewReader(body)
-	err = json.NewDecoder(rdr).Decode(&response)
-	return response, body, err
+	err = json.Unmarshal(buf.Bytes(), &response)
+	return response, err
+}
+
+// ImageInspectWithRaw returns the image information and its raw representation.
+//
+// Deprecated: Use [ImageInspect] instead.
+func (cli *Client) ImageInspectWithRaw(ctx context.Context, imageID string) (image.InspectResponse, []byte, error) {
+	var buf bytes.Buffer
+	resp, err := cli.ImageInspect(ctx, imageID, ImageInspectWithRawResponse(&buf))
+	if err != nil {
+		return resp, nil, err
```

Comment:
Nit; better to be  explicit that we're not returning anything here (yeah, we're doing that in many  places,  but as  this is modified in this PR);

```suggestion
		return image.InspectResponse{}, nil, err
```